rasa/nlu/model.py

```diff
@@ -307,25 +313,63 @@ def load(
             model_dir: The path of the model to load
             component_builder: The
                 :class:`rasa.nlu.components.ComponentBuilder` to use.
+            new_config: Optional new config to use for the new epochs.
+            finetuning_epoch_fraction: Value to multiply all epochs by.
 
         Returns:
             An interpreter that uses the loaded model.
         """
-
         model_metadata = Metadata.load(model_dir)
 
+        if new_config:
+            Interpreter._update_epochs_from_new_config(
+                model_metadata, new_config, finetuning_epoch_fraction
+            )
+
         Interpreter.ensure_model_compatibility(model_metadata)
-        return Interpreter.create(model_metadata, component_builder, skip_validation)
+        return Interpreter.create(
+            model_metadata,
+            component_builder,
+            skip_validation,
+            should_finetune=new_config is not None,
+        )
+
+    @staticmethod
+    def _update_epochs_from_new_config(
+        model_metadata: Metadata,
+        new_config: Optional[Dict] = None,
+        finetuning_epoch_fraction: float = 1.0,
+    ):
```

Comment:
```suggestion
    ) -> None:
```

Reply:
optional: The code would feel a bit more intuitive to me if the function would return the updated `Metadata` object as it would put the focus on what part is updated exactly.
```suggestion
    ) -> Metadata:
```

Reply:
I personally try to avoid methods which both mutate and return a value. I find it makes it unclear whether mutation is occurring. I could change the method to create a new metadata, although maybe that would be a pain. WDYT?

Reply:
> I could change the method to create a new metadata, although maybe that would be a pain. WDYT?

Mhm, it would be probably a `deepcopy` 🤔  Do you think naming it `_update_epochs_in_new_config` or `_update_config_with_epochs` would make the inplace change more explicit?